providers/runtime.go

```diff
@@ -585,6 +585,7 @@ func (r *Runtime) lookupResourceProvider(resource string) (*ConnectedProvider, *
 	}
 
 	if info.Provider != providerConn && !stringx.Contains(crossProviderList, info.Provider) {
+		log.Error().Str("infoProvider", info.Provider).Str("connectionProvider", providerConn).Msg("mismatch between expected and received provider, ignoring provider")
```

Comment:
would be helpful to log what is not matching, eg. via debug log

Reply:
I found this message super helpful in understanding what is not matching. The corssProviderList is static, so this message only happens when the two are different.

Reply:
we are on the same page, just missed that providerConn == info.Provider